shared/src/common/enums/user-details/user-details-validation-rule.enum.ts

```diff
@@ -0,0 +1,14 @@
+const today = new Date();
+const MIN_DATE = new Date(
+  today.setFullYear(today.getFullYear() - 100),
+).toDateString();
+
+const UserDetailsValidationRule = {
+  FULL_NAME_MIN_LENGTH: 2,
+  FULL_NAME_MAX_LENGTH: 40,
+  FULL_NAME_PATTERN: /^[ A-Za-z-'`]*$/,
+  MIN_DATE: MIN_DATE,
+  MAX_DATE: new Date().toDateString(),
```

Comment:
it should be a hardcoded value after the deployment. let's avoid max and min date validation for now